task8/src/main/java/AdCollection/AdService.java

```diff
@@ -0,0 +1,158 @@
+package AdCollection;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import com.google.gson.Gson;
+
+public class AdService {
+    private List<Ad> list;
+
+    public static AdService INSTANCE = null;
```

Comment:
1) Place static members above instance members.
2) Seems like you are implementing the singleton pattern but in its non-threadsafe variant. The following straightforward code will work better for a client-server app
```
public static finsal AdService INSTANCE = new AdService();
```
You can read more on implementing singletoins e.g. here https://habr.com/ru/post/27108/